agent/agent-core/src/main/java/org/bithon/agent/configuration/source/EnvironmentSource.java

```diff
@@ -36,8 +37,9 @@ public static PropertySource build(String envPrefix) {
             if (name.startsWith(envPrefix) && !value.isEmpty()) {
                 name = name.substring(envPrefix.length())
                            // For env, the underscore is used as a replacement of '.' character,
-                           // Here we need to convert these characters back
-                           .replace('_', '.');
+                           // Here we need to convert these characters back.
+                           // Additionally, as the env variables are in uppercase, we need to convert them to lowercase.
+                           .replace('_', '.').toLowerCase(Locale.ENGLISH);
```

Comment:
Use `toLowerCase` to convert environment variable values from upper case to lower case may be not always correct.

Currently, any configuration class can be deserialized from external configuration source like file or environment variables, these classes may contain fields that are using camel style naming conventions.

For example, in `org.bithon.agent.observability.tracing.config.TraceConfig`, there's a field naming as `traceResponseHeader`. Also existing test case does not cover such scenario.

So to support the uppcase of the environment variables is a little bit tricky now. I have not come up with any idea to support it.

Reply:
After reading the relevant code, I had a general understanding that the scope involved exceeded my previous estimation.

Reply:
Can add a property like `camelNamingStyleProp` to the `TestProp` class in the test file and improve the test case `test_Environment` to verify the deserialization for such field.

This makes sure new changes in this PR is compatible with such case. The comment below may be one way to resolve such problem.